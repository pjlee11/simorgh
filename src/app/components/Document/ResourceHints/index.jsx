import React, { Fragment } from 'react';

const ResourceHints = () => (
  <Fragment>
    <link
      rel="preconnect"
      href="https://ichef.bbci.co.uk"
      crossOrigin="anonymous"
    />
    <link
      rel="preconnect"
      href="https://static.bbci.co.uk"
      crossOrigin="anonymous"
    />
    <link
      rel="preconnect"
      href="https://gel.files.bbci.co.uk"
      crossOrigin="anonymous"
    />
    <link rel="dns-prefetch" href="https://ichef.bbci.co.uk" />
    <link rel="dns-prefetch" href="https://static.bbci.co.uk" />
    <link rel="dns-prefetch" href="https://gel.files.bbci.co.uk" />
  </Fragment>
);

export default ResourceHints;
