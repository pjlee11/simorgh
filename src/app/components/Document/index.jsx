import React from 'react';
import ResourceHints from './ResourceHints';
import '../../lib/globalStyles';
import { C_POSTBOX } from '../../lib/constants/styles';

/* eslint-disable react/prop-types */
const Document = ({ assets, app, data, styleTags, helmet }) => {
  const htmlAttrs = helmet.htmlAttributes.toComponent();
  const meta = helmet.meta.toComponent();
  const title = helmet.title.toComponent();
  const links = helmet.link.toComponent();
  const serialisedData = JSON.stringify(data);
  const scripts = assets.map(asset => (
    <script
      crossOrigin="anonymous"
      key={asset}
      type="text/javascript"
      src={asset}
      defer
    />
  ));

  return (
    <html lang="en-GB" {...htmlAttrs}>
      <head>
        <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
        <meta charSet="utf-8" />
        <meta name="robots" content="noindex,nofollow" />
        <meta name="theme-color" content={C_POSTBOX} />
        {meta}
        <link rel="shortcut icon" href="/favicon.ico" type="image/x-icon" />
        <link rel="manifest" href="/manifest.json" />
        <ResourceHints />
        {title}
        {links}
        {styleTags}
      </head>
      <body>
        {/* eslint-disable react/no-danger */
        /* disabling the rule that bans the use of dangerouslySetInnerHTML until a more appropriate implementation can be implemented */}
        <div id="root" dangerouslySetInnerHTML={{ __html: app }} />
        <script
          dangerouslySetInnerHTML={{
            __html: `window.SIMORGH_DATA=${serialisedData}`,
          }}
        />
        {scripts}
      </body>
    </html>
  );
};

export default Document;
