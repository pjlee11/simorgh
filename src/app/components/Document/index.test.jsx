import React from 'react';
import { shallowRender } from '../../helpers/tests/testHelpers';
import Document from './index';

describe('Document', () => {
  const assets = ['http://example.com/file.js'];
  const data = { test: 'data' };

  const mockHelmetToComponent = element => ({
    toComponent: jest.fn().mockImplementation(() => element),
  });

  const helmet = {
    htmlAttributes: mockHelmetToComponent({
      lang: 'test',
    }),
    link: mockHelmetToComponent(
      <link
        rel="canonical"
        href="https://www.bbc.com/news/articles/c9rpqy7pmypo"
      />,
    ),
    meta: mockHelmetToComponent(
      <meta
        name="viewport"
        content="width=device-width, initial-scale=1, minimum-scale=1"
      />,
    ),
    title: mockHelmetToComponent(<title>Test title</title>),
  };
  const styleTags = <style>{'html { color: red; }'}</style>;

  const shallowDocument = shallowRender(
    <Document
      assets={assets}
      app={'<h1>App!</h1>'}
      data={data}
      helmet={helmet}
      styleTags={styleTags}
    />,
  );

  it('should render correctly', () => {
    expect(shallowDocument).toMatchSnapshot();
  });
});
