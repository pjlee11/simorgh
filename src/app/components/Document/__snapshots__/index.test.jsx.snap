// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document should render correctly 1`] = `
<html
  lang="test"
>
  <head>
    <meta
      content="IE=edge"
      httpEquiv="X-UA-Compatible"
    />
    <meta
      charSet="utf-8"
    />
    <meta
      content="noindex,nofollow"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <link
      href="/manifest.json"
      rel="manifest"
    />
    <ResourceHints />
    <title>
      Test title
    </title>
    <link
      href="https://www.bbc.com/news/articles/c9rpqy7pmypo"
      rel="canonical"
    />
    <style>
      html { color: red; }
    </style>
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1>App!</h1>",
        }
      }
      id="root"
    />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.SIMORGH_DATA={\\"test\\":\\"data\\"}",
        }
      }
    />
    <script
      crossOrigin="anonymous"
      defer={true}
      src="http://example.com/file.js"
      type="text/javascript"
    />
  </body>
</html>
`;
